pkg/workspace/plugins.go

```diff
@@ -330,6 +330,28 @@ func GetPluginPath(kind PluginKind, name string, version *semver.Version) (strin
 		return "", path, nil
 	}
 
+	// At some point in the future, language plugins will be located in the plugin cache, just like regular plugins
+	// (see pulumi/pulumi#956 for some of the reasons why this isn't the case today). For now, they ship next to the
+	// `pulumi` binary. While we encourage this folder to be on the $PATH (and so the check above would have found
+	// the language plugin) it's possible someone is running `pulumi` with an explicit path on the command line or
+	// has done symlink magic such that `pulumi` is on the path, but the language plugins are not. So, if possible,
+	// look next to the instance of `pulumi` that is running to find this language plugin.
+	if kind == LanguagePlugin {
+		exePath, exeErr := os.Executable()
+		if exeErr == nil {
+			fullPath, fullErr := filepath.EvalSymlinks(exePath)
+			if fullErr == nil {
+				candidate := filepath.Join(filepath.Dir(fullPath), filename)
+				if _, err := os.Stat(candidate); err == nil {
```

Comment:
`!os.IsNotExist(err)`?

Also should we try and verify that this thing is executable before we claim victory?

Reply:
> `!os.IsNotExist(err)`?

Not 100% sure here.  I'm checking to ensure that err was nil, so any error from stat would cause us to ignore this candidate.  I can't think of a way to slice the error space that makes it possible to do something reasonable for some class of errors but not others.

> Also should we try and verify that the candidate is executable before we claim victory?

Sure.